lib/checkstl.cpp

```diff
@@ -1899,15 +1899,19 @@ void CheckStl::string_c_str()
     const SymbolDatabase* symbolDatabase = mTokenizer->getSymbolDatabase();
 
     // Find all functions that take std::string as argument
-    std::multimap<const Function*, int> c_strFuncParam;
+    struct StrArg {
+        nonneg int n; // cppcheck-suppress unusedStructMember // FP used through iterator/pair
+        std::string argtype; // cppcheck-suppress unusedStructMember
+    };
+    std::multimap<const Function*, StrArg> c_strFuncParam;
     if (printPerformance) {
         for (const Scope &scope : symbolDatabase->scopeList) {
             for (const Function &func : scope.functionList) {
-                int numpar = 0;
+                nonneg int numpar = 0;
                 for (const Variable &var : func.argumentList) {
                     numpar++;
-                    if (var.isStlStringType() && (!var.isReference() || var.isConst()))
-                        c_strFuncParam.insert(std::make_pair(&func, numpar));
+                    if ((var.isStlStringType() || var.isStlStringViewType()) && (!var.isReference() || var.isConst()))
+                        c_strFuncParam.insert(std::make_pair(&func, StrArg{ numpar, var.getTypeName() }));
```

Comment:
This should work: `c_strFunParam.emplace(&func, StrArg{ numpar, var.getTypeName() });`